### ISSUE TITLE
Question: One entry, multiple URLs

### ISSUE THREAD

**Issue by tbradsha** [keeweb/keeweb#941]
I know this same question has been asked a few times, but none of them seem to have any working examples. I've tried various things and can't yet get it to work. Can you clarify if the following is expected to work at this time or if it requires waiting for #433?

I am using the Mac Desktop app and Chrome (I've also tried with Safari). I have two websites I use that share the same login information: domainone.com and domaintwo.com. The login URL for both is like this (where some_random_secret is different every load):

https://ident.domainone.com/oauth2/authorization?client_secret=some_random_secret
https://ident.domaintwo.com/oauth2/authorization?client_secret=some_random_secret

The title for each is as follows:

Sign in to domainone.com
Sign in to domaintwo.com

No matter what I put in custom fields, I can't seem to get the second domain to match for autotype. Can you give an example of what should be in the custom field so as to match both domains?

**Comment by antelle** [comment]
Hi, this is not related to the issue you mentioned.
It depends on what happens during auto-type. Could you please check if the app can detect the url? It will be there in logs, if so.

**Comment by tbradsha** [comment]
The app successfully detects the URL as well as the window title. Here are the logs (slightly modified for privacy) for domainone.com, which works fine:

```
[DEBUG] 2018-06-11T18:24:02.992Z [auto-type] Auto type event null
[DEBUG] 2018-06-11T18:24:02.992Z [auto-type] Get window title
[INFO ] 2018-06-11T18:24:04.088Z [launcher] spawn C:\Program Files\KeeWeb\resources\helper\win32\KeeWebHelper.exe: 0, 1090ms
Sign in to domainone.com
https://ident.domainone.com/oauth2/authorization?client_secret=some_random_secret
[DEBUG] 2018-06-11T18:24:04.088Z [auto-type] Window title Sign in to domainone.com https://ident.domainone.com/oauth2/authorization?client_secret=some_random_secret
[DEBUG] 2018-06-11T18:24:04.088Z [auto-type] Hide window
[DEBUG] 2018-06-11T18:24:04.088Z [auto-type] Start {USERNAME}{TAB}{PASSWORD}{ENTER}
[DEBUG] 2018-06-11T18:24:04.088Z [auto-type] Parsed [op:USERNAME,op:TAB,op:PASSWORD,op:ENTER]
[DEBUG] 2018-06-11T18:24:04.088Z [auto-type] Resolved [*,key:tab,[*],key:enter]
[INFO ] 2018-06-11T18:24:04.289Z [launcher] spawn C:\Program Files\KeeWeb\resources\helper\win32\KeeWebHelper.exe: 0, 192ms
[DEBUG] 2018-06-11T18:24:04.289Z [auto-type] Complete 193ms
```


And here are the logs for domaintwo.com, which doesn't autodetect and type, but requires me to select the entry from the full list of options:

```
[DEBUG] 2018-06-11T18:24:40.935Z [auto-type] Auto type event null
[DEBUG] 2018-06-11T18:24:40.936Z [auto-type] Get window title
[INFO ] 2018-06-11T18:24:41.993Z [launcher] spawn C:\Program Files\KeeWeb\resources\helper\win32\KeeWebHelper.exe: 0, 1063ms
Sign in to domaintwo.com
https://ident.domaintwo.com/oauth2/authorization?client_secret=some_random_secret
[DEBUG] 2018-06-11T18:24:41.993Z [auto-type] Window title Sign in to domaintwo.com https://ident.domaintwo.com/oauth2/authorization?client_secret=some_random_secret
[DEBUG] 2018-06-11T18:24:46.605Z [auto-type] Entry selected {"id":"base64_id","uuid":"base64_uuid"}
[DEBUG] 2018-06-11T18:24:46.610Z [auto-type] Hide window
[DEBUG] 2018-06-11T18:24:46.735Z [auto-type] Start {USERNAME}{TAB}{PASSWORD}{ENTER}
[DEBUG] 2018-06-11T18:24:46.735Z [auto-type] Parsed [op:USERNAME,op:TAB,op:PASSWORD,op:ENTER]
[DEBUG] 2018-06-11T18:24:46.735Z [auto-type] Resolved [*,key:tab,[*],key:enter]
[INFO ] 2018-06-11T18:24:46.932Z [launcher] spawn C:\Program Files\KeeWeb\resources\helper\win32\KeeWebHelper.exe: 0, 196ms
[DEBUG] 2018-06-11T18:24:46.932Z [auto-type] Complete 197ms
```

**Comment by antelle** [comment]
Okay, so it got the url :)
Then it means, you can add these two urls in your entry custom fields, e.g.:
`website1`: `https://ident.domaintwo.com/`,
`website2`: `https://ident.domainone.com/`

**Comment by tbradsha** [comment]
Yes, I've done that already, but it isn't detecting:

> User: xxx
> Pass: xxx
> Website: https://ident.domainone.com/
> Website1: https://ident.domainone.com/
> Website2: https://ident.domaintwo.com/

Does the URL path matter? I've even tried with the following without success:

> Website1: https://ident.domainone.com/oauth2/authorization
> Website2: https://ident.domaintwo.com/oauth2/authorization

**Comment by antelle** [comment]
Yes, it must be the full url. Maybe your url has some params in it?

**Comment by tbradsha** [comment]
Yes, it has this parameter, where some_random_secret changes every load:
`?client_secret=some_random_secret`

How do I handle this?

**Comment by antelle** [comment]
There's no solution for now. Let's leave it open to think.

**Comment by tbradsha** [comment]
I think if the parameters are ignored, it would be a solution (match on the domain and path portion only).

All this said, shouldn't this detect based on window title as well? Why does this not work? Or perhaps I misunderstood how this should work.

**Comment by antelle** [comment]
It does match by title, domain, and so on, but only in the main website and entry title. We shouldn't search window title and apply these fuzzy url matching rules to other fields, it can give a lot of false positives. However we always can show an entry selector, for example, if the entry was found through this. I don't know for now, it needs more thinking.

**Comment by tbradsha** [comment]
As a thought, can we use a custom field such as "include" or "match" that will contain a regex value to be tested against the URL and/or window title?

**Comment by antelle** [comment]
@tbradsha that's what should be done with auto-type windows configuration item.

**Comment by tbradsha** [comment]
I don't currently see any auto-type solution for matching multiple URLs with dynamic params.

If there were a special "match" field I could then put this in the field:

```
https://ident.domain(one|two).com
```

If KeeWeb allowed for a single regex field (or allowed regex in the Website field) to match the site title/window title/URL, that'd be great. I understand not wanting to match all fields.

If there's already a different solution, though, can you clarify? I'm not quite sure I understand your last comment.

**Comment by bdeeming** [comment]
An additional reference implementation that appears to work OK: [Keepass2Android](https://play.google.com/store/apps/details?id=keepass2android.keepass2android) uses additional fields with a specific name structure, one field per URL (or app URI/identifier in its case).
I imagine the primary use case (for myself it is at least), is for most of the URL to be swapped out (rather than a small permutation of the original domain, thus a list of full URL's may be easier to work with. Particularly for tools to add to it in an automated fashion, and also for making links that can be clicked to go to the URL's from the app itself.

**Comment by antelle** [comment]
@tbradsha I mean this thing: https://keepass.info/%0D/help/base/autotype.html
We need to port this configuration option to keeweb.

**Comment by tbradsha** [comment]
Ah, didn't know about that. This would be great if it were implemented!

**Comment by antelle** [comment]
The bot was misconfigured, sorry for a wrong notification.
